lib/core/src/animation/animator/easing.rs

```diff
@@ -0,0 +1,92 @@
+//! This file provides the implementation of EasingAnimator.
+
+use super::ContinuousAnimator;
+use crate::animation::easing::FnEasing;
+use crate::control::event_loop::EventLoop;
+use crate::math::utils::linear_interpolation;
+use crate::math::utils::Interpolable;
+
+use nalgebra::clamp;
+use std::rc::Rc;
+use std::cell::RefCell;
+
+
+
+// ===============================
+// === EasingAnimationCallback ===
+// ===============================
+
+pub trait EasingAnimationCallback<T> = FnMut(T) + 'static;
+
+
+
+// ==========================
+// === EasingAnimatorData ===
+// ==========================
+
+struct EasingAnimatorData<T:Interpolable<T>+'static> {
```

Comment:
Because you used `+'static` here - it propagated all the way down to `EasingAnimator` and also to `impls` below. I mentioned some time ago that almost every usage of `'static` in other contexts than `F:Fn() + 'static` when `F` gets Boxed is an orange lamp. Sure, compiler tells you that this is a possible solution, but in 90% of cases this is ugly solution. Like here. Please remove it and do not use `'static` if not really, really needed.

Reply:
I removed most of them, but L46 `impl<T:Interpolable<T>+'static> EasingAnimator<T>` still seemed necessary. I really don't understand why it needs an explicit lifetime because T is moved as a copy and not as a reference.